test/cli/other_test.py

```diff
@@ -3654,3 +3654,67 @@ def test_debug_syntaxerror_c(tmp_path):
     assert stderr.splitlines() == [
         "{}:2:1: error: Code 'template<...' is invalid C code. [syntaxError]".format(test_file)
     ]
+
+
+def test_defined_configs(tmp_path):
+    test_file = tmp_path / 'test.c'
+    with open(test_file, "w") as f:
+        f.write(
+            """
+int main() {
+#ifdef A1
+#endif
+#ifdef A2
+#endif
+#ifdef A3
+#endif
+return 0;
+}
+""")
+    args = [
+        '-DAAA',
+        '--template=simple',
+        str(test_file)
+    ]
+    exitcode, stdout, stderr = cppcheck(args)
+    assert exitcode == 0, stderr
+    assert stdout == ('Checking %s ...\n'
+                      'Checking %s: AAA=1...\n' % (test_file, test_file))
+    args = [
+        '--max-configs=2',
+        '-DAAA',
+        '--template=simple',
+        str(test_file)
+    ]
+    exitcode, stdout, stderr = cppcheck(args)
+    assert exitcode == 0, stderr
+    assert stdout == ('Checking %s ...\n'
+                      'Checking %s: AAA=1...\n'
+                      'Checking %s: AAA=1;A1...\n' % (test_file, test_file, test_file))
+
+    args = [
+        '--force',
+        '-DAAA',
+        '--template=simple',
+        str(test_file)
+    ]
+    exitcode, stdout, stderr = cppcheck(args)
+    assert exitcode == 0, stderr
+    assert stdout == ('Checking %s ...\n'
+                      'Checking %s: AAA=1...\n'
+                      'Checking %s: AAA=1;A1...\n'
+                      'Checking %s: AAA=1;A2...\n'
+                      'Checking %s: AAA=1;A3...\n' % (test_file, test_file, test_file, test_file, test_file))
+
+    args = [
```

Comment:
This is a promising start. I would also like to see tests that check what happens when a `--project=compile_commands.json` is used.